File: AdvantageFramework.Core/Database/Entities/AlertAttachment.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdvantageFramework.Core.Database.Entities
{
    [Table("ALERT_ATTACHMENT")]
    public partial class AlertAttachment
    {
        #region Constants



        #endregion

        #region Enum

        public enum Properties
        {
            TBD
        }


        #endregion

        #region Variables



        #endregion

        #region Properties

        [Key]
        [Column("ATTACHMENT_ID")]
        public int AttachmentId { get; set; }
        [Column("ALERT_ID")]
        public int AlertId { get; set; }
        [Required]
        [Column("USER_CODE")]
        [StringLength(100)]
        public string UserCode { get; set; }
        [Column("GENERATED_DATE", TypeName = "smalldatetime")]
        public DateTime GeneratedDate { get; set; }
        [Required]
        [Column("EMAILSENT")]
        public bool? Emailsent { get; set; }
        [Column("DOCUMENT_ID")]
        public int DocumentId { get; set; }
        [Column("USER_CODE_CP")]
        public int? UserCodeCp { get; set; }
        [Column("THUMBNAIL")]
        public byte[] Thumbnail { get; set; }

        [ForeignKey(nameof(AlertId))]
        [InverseProperty("AlertAttachments")]
        public virtual Alert Alert { get; set; }
        [ForeignKey(nameof(DocumentId))]
        [InverseProperty("AlertAttachments")]
        public virtual Document Document { get; set; }

        #endregion

        #region Methods



        #endregion
    }
}

File: AdvantageFramework/Database/Entities/AlertAttachment.vb
Namespace Database.Entities

	<Table("ALERT_ATTACHMENT")>
	Public Class AlertAttachment
		Inherits BaseClasses.Entity

#Region " Constants "



#End Region

#Region " Enum "

		Public Enum Properties
			ID
			AlertID
			UserCode
			GeneratedDate
			HasEmailBeenSent
			DocumentID
			ClientContactID
			Alert

		End Enum

#End Region

#Region " Variables "



#End Region

#Region " Properties "

		<Key>
		<System.ComponentModel.DataAnnotations.Schema.DatabaseGeneratedAttribute(ComponentModel.DataAnnotations.Schema.DatabaseGeneratedOption.Identity)>
		<Required>
		<Column("ATTACHMENT_ID")>
		Public Property ID() As Integer
		<Required>
		<Column("ALERT_ID")>
		Public Property AlertID() As Integer
		<Required>
		<MaxLength(100)>
		<Column("USER_CODE", TypeName:="varchar")>
		Public Property UserCode() As String
		<Required>
		<Column("GENERATED_DATE")>
		Public Property GeneratedDate() As Date
		<Required>
		<Column("EMAILSENT")>
		Public Property HasEmailBeenSent() As Boolean
		<Required>
		<Column("DOCUMENT_ID")>
		Public Property DocumentID() As Integer
		<Column("USER_CODE_CP")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ClientContactID() As Nullable(Of Integer)
        <Column("THUMBNAIL")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Thumbnail() As Byte()

        Public Overridable Property Alert As Database.Entities.Alert

#End Region

#Region " Methods "

        Public Overrides Function ToString() As String

            ToString = Me.ID

        End Function

#End Region

	End Class

End Namespace
